File: src/main/java/com/github/graqr/threshr/model/redsky/products/pdp/client/ValuePreparedList.java
package com.github.graqr.threshr.model.redsky.products.pdp.client;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record ValuePreparedList(
        String description,
        @JsonProperty("serving_size") String servingSize,
        @JsonProperty("serving_size_unit_of_measurement") String servingSizeUnitOfMeasurement,
        @JsonProperty("servings_per_container") String servingsPerContainer,
        List<Nutrient> nutrients) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/product/ValuePreparedList.kt
package com.graqr.threshr.model.redsky.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class ValuePreparedList(
    val description: String,
    @param:JsonProperty("serving_size") val servingSize: String,
    @param:JsonProperty("serving_size_unit_of_measurement") val servingSizeUnitOfMeasurement: String,
    @param:JsonProperty("servings_per_container") val servingsPerContainer: String,
    val nutrients: List<Nutrient>
) {}
